tests/system-tests/internal/supporttools/traceroute.go

```diff
@@ -0,0 +1,479 @@
+	targetPort,
+	searchString string) error {
+	timeout := 3 * time.Minute
+	err := wait.PollUntilContextTimeout(
+		context.TODO(),
+		3*time.Second,
+		timeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			result, err := sendProbesAndCheckOutput(
+				trPodObj,
+				targetIP,
+				targetPort,
+				searchString)
+
+			if err == nil && result {
+				return true, nil
+			}
+
+			return false, nil
+		})
+
+	if err != nil {
+		glog.V(100).Infof("expected string was not found in the traceroute output; %v", err)
+
+		return fmt.Errorf("expected string was not found in the traceroute output; %w", err)
+	}
+
+	return nil
+}
+
+func defineTraceRouteContainer(cImage string) *pod.ContainerBuilder {
+	cName := "traceroute"
+
+	cCmd := []string{"/bin/bash", "-c", "sleep INF"}
+
+	glog.V(100).Infof("Creating container %q", cName)
+
+	deployContainer := pod.NewContainerBuilder(cName, cImage, cCmd)
+
+	glog.V(100).Infof("Defining SecurityContext")
+
+	var trueFlag = true
+
+	userUID := new(int64)
+
+	*userUID = 0
+
+	securityContext := &corev1.SecurityContext{
+		RunAsUser:  userUID,
+		Privileged: &trueFlag,
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{
+				"SETFCAP",
+				"CAP_NET_RAW",
+				"CAP_NET_ADMIN",
+			},
+		},
+	}
+
+	glog.V(100).Infof("Setting SecurityContext")
+
+	deployContainer = deployContainer.WithSecurityContext(securityContext)
+
+	glog.V(100).Infof("Dropping ALL security capability")
+
+	deployContainer = deployContainer.WithDropSecurityCapabilities([]string{"ALL"}, true)
+
+	glog.V(100).Infof("Enable TTY and Stdin; needed for immediate log propagation")
+
+	deployContainer = deployContainer.WithTTY(true).WithStdin(true)
+
+	glog.V(100).Infof("%q container's  definition:\n%#v", cName, deployContainer)
+
+	return deployContainer
+}
+
+func defineDeployment(
+	apiClient *clients.Settings,
+	containerConfig *corev1.Container,
+	deployName, deployNs, saName string,
+	scheduleOnHosts []string,
+	deployLabels map[string]string) *deployment.Builder {
+	glog.V(100).Infof("Defining deployment %q in %q ns", deployName, deployNs)
+
+	nodeAffinity := corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "kubernetes.io/hostname",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   scheduleOnHosts,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	trDeployment := deployment.NewBuilder(apiClient, deployName, deployNs, deployLabels, *containerConfig)
+
+	glog.V(100).Infof("Assigning ServiceAccount %q to the deployment", saName)
+
+	trDeployment = trDeployment.WithServiceAccountName(saName)
+
+	glog.V(100).Infof("Setting Replicas count")
+
+	replicasCnt := len(scheduleOnHosts)
```

Comment:
I'd rather expose replica count as a parameter to cover a use case where a pod(s) can ran on multiple nodes but different amount of replicas is required (for e.g. a pod could run on node-x, node-y, node-z, but app requires only 2 instances).
Anyway that's only a suggestion so leaving it up to U

Reply:
I'm sorry, but I'm not sure it's a good idea. different nodes/pods count is not exactly our case.
Also, the question is, why provide more nodes than you need pods?
In our case, for both tcpdump and traceroute scenarios, we provide precisely the nodes list where we need to deploy the support tools pods; I mean, we need to monitor a specific node => we need a pod on this node.
I would leave it like this right now if you don't mind.
If we have any future use cases with different requirements, we can modify it. But without a specific proposal for different nodes and pod counts, it's hard to understand how it would be better to change it.

Reply:
> In our case, for both tcpdump and traceroute scenarios, we provide precisely the nodes list where we need to deploy the support tools pods; I mean, we need to monitor a specific node => we need a pod on this node.
I would leave it like this right now if you don't mind.

My point is that it's better leave any computations of the replicaCount outside of the method and doesn't make it  dependent on some other parameter. But if U like to leave as is - that works for me as well